src/lib/CollisionPrevention/CMakeLists.txt

```diff
@@ -33,3 +33,18 @@
 
 px4_add_library(CollisionPrevention CollisionPrevention.cpp)
 target_compile_options(CollisionPrevention PRIVATE -Wno-cast-align) # TODO: fix and enable
+
+px4_add_gtest(SRC CollisionPreventionTest.cpp LINKLIBS CollisionPrevention px4_daemon
```

Comment:
It's worrying to me that we have so many dependencies for a single unit test.

Reply:
These are the dependencies that get pulled in anyway by the build system. I just had to list them manually again to put the stubs afterwards, otherwise we get linking errors (since linker only looks for symbols in later libraries).

I'd love for there to be a better solution to this, but I think it would require manual dependency management instead of letting CMake do it for us.